include/zephyr/posix/pthread.h

```diff
@@ -432,7 +432,7 @@ int pthread_attr_setstack(pthread_attr_t *attr, void *stackaddr,
 #ifdef CONFIG_PTHREAD_IPC
 int pthread_once(pthread_once_t *once, void (*initFunc)(void));
 #endif
-void pthread_exit(void *retval);
+void pthread_exit(void *retval) FUNC_NORETURN;
```

Comment:
Shouldn't this be in front to follow the convention?

```suggestion
FUNC_NORETURN void pthread_exit(void *retval);
```

Reply:
Sure. I can do that if you prefer. Although function attributes can go either way, conventionally.

https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html